src/AEG/AEGUtils.ts

```diff
@@ -65,30 +65,68 @@ export function shapesIntersect(
-    let cornerOutOfEllipse = false;
-    let val: number;
+    const wideCoords = getWidestCoordinates(ellipse);
 
-    for (let i = 0; i < 4; i++) {
-        val = signedDistanceFromEllipse(ellipse, corners[i]);
-        if (val < 0) {
-            //The corner is within the ellipse
-            cornerInEllipse = true;
-        } else if (val > 0) {
-            //The corner is outside the ellipse
-            cornerOutOfEllipse = true;
-        }
+    const leftBound = ellipse.radiusX > rectangle.width ? wideCoords[3].x : corners[0].x;
+    const rightBound = ellipse.radiusX > rectangle.width ? wideCoords[1].x : corners[1].x;
+    const topBound = ellipse.radiusY >= rectangle.height ? wideCoords[0].y : corners[0].y;
+    const bottomBound = ellipse.radiusY >= rectangle.height ? wideCoords[2].y : corners[2].y;
+
+    //Check for vertical edge intersection
+    const leftEdgeUpperCurve = getCurvePoint(ellipse, corners[0].x, 1);
+    const rightEdgeUpperCurve = getCurvePoint(ellipse, corners[1].x, 1);
+    const leftEdgeLowerCurve = getCurvePoint(ellipse, corners[0].x, -1);
+    const rightEdgeLowerCurve = getCurvePoint(ellipse, corners[1].x, -1);
+
+    if (
+        (topBound < leftEdgeUpperCurve && leftEdgeUpperCurve < bottomBound) ||
+        (topBound < leftEdgeLowerCurve && leftEdgeLowerCurve < bottomBound) ||
+        (topBound < rightEdgeUpperCurve && rightEdgeUpperCurve < bottomBound) ||
+        (topBound < rightEdgeLowerCurve && rightEdgeLowerCurve < bottomBound)
+    ) {
+        return true;
+    }
+
+    //Check for horizontal edge intersection
+    const a = -1 / Math.pow(ellipse.radiusX, 2);
+    const b = (2 * ellipse.center.x) / Math.pow(ellipse.radiusX, 2);
+
+    //c with height of the top edge
+    const cTop =
+        -Math.pow(ellipse.center.x, 2) / Math.pow(ellipse.radiusX, 2) +
+        1 -
+        Math.pow((corners[0].y - ellipse.center.y) / ellipse.radiusY, 2);
+
+    //c with height of the bottom edge
+    const cBottom =
+        -Math.pow(ellipse.center.x, 2) / Math.pow(ellipse.radiusX, 2) +
+        1 -
+        Math.pow((corners[2].y - ellipse.center.y) / ellipse.radiusY, 2);
+
+    const topX1 = (-b + Math.sqrt(Math.pow(b, 2) - 4 * a * cTop)) / (2 * a);
+    const topX2 = (-b - Math.sqrt(Math.pow(b, 2) - 4 * a * cTop)) / (2 * a);
+    const bottomX1 = (-b + Math.sqrt(Math.pow(b, 2) - 4 * a * cBottom)) / (2 * a);
+    const bottomX2 = (-b - Math.sqrt(Math.pow(b, 2) - 4 * a * cBottom)) / (2 * a);
+
+    //Horizontal intersection if the intersection point is within the bounds of the ellipse
+    if (
+        (!isNaN(topX1) && leftBound < topX1 && topX1 < rightBound) ||
+        (!isNaN(topX2) && leftBound < topX2 && topX2 < rightBound) ||
+        (!isNaN(bottomX1) && leftBound < bottomX1 && bottomX1 < rightBound) ||
+        (!isNaN(bottomX2) && leftBound < bottomX2 && bottomX2 < rightBound)
+    ) {
+        return true;
     }
 
-    return cornerInEllipse && cornerOutOfEllipse;
+    return false;
```

Comment:
`if(cond){return true;} return false;` should always just be written `return cond;`